cantools/database/can/c_source.py

```diff
@@ -691,6 +761,52 @@ def maximum_value(self):
         else:
             return ((2 ** self.length) - 1)
 
+    def get_json_formatting(self):
+        # This returns a string that is used to format this signal
+        # into a JSON string. E.G. for a signal FanSpeed of type Float:
+        #  "FanSpeed": %15.3e,
+        # TODO add other types here?
+        if self.is_float:
+            return "%15.3e"
+        else:
+            return "%d"
+
+    def get_json_length(self):
+        # This returns the number of bytes required to represent this signal
+        # in JSON, assuming worst-case float representation, escaped quotation marks, etc.
+        # E.G. \"FanSpeed\": 1234567.123e-12,
+        # Signal name: 8
+        # Quotation: 4
+        # Colon and space: 2
+        # Float representation: 15 (1234567.123e-12)
+        # Comma: 1
+        result = len(self.exported_name)
+        result += 7 # Escaped quotation, colon, space, comma
+        if self.is_float:
+            result += 15
+        else:
+            # TODO: Fix these
```

Comment:
```
>>> for bit_len in (8, 16, 32, 64):
...     print("max len for int{}_t:".format(bit_len), len(str(-(2**(bit_len-1)))))
...     print("max len for uint{}_t:".format(bit_len), len(str(2**bit_len-1)))
...
max len for int8_t: 4
max len for uint8_t: 3
max len for int16_t: 6
max len for uint16_t: 5
max len for int32_t: 11
max len for uint32_t: 10
max len for int64_t: 20
max len for uint64_t: 20
```

I'd suggest using that code to build a module level lookup table, and then doing a lookup on the type name here, rather than hardcoding the magic numbers in an if statement.

Reply:
I removed those, and instead rely on the individual-signal JSON decode buffer being big enough to hold any single decoded signal name and value, rather than trying to calculate whether the signal+value would be too big for the buffer. The maximum practical length of a signal name is 32 characters, and enum values are max 128 characters, the 256-character JSON output buffer should suffice.